[BLOG] The road to Scaphandre v1.0 : Challenges and improvements to come on IT energy consumption evaluation (#245)
cncf/tag-env-sustainability: website/content/blog/2023-10-the-road-to-scaphandre-v1.0-challenges-and-improvements-to-come-on-IT-energy-consumption-evaluation.md

---

> title: The road to Scaphandre v1.0 : Challenges and improvements to come on IT energy consumption evaluation

Reviewer: ```suggestion
title: The road to Scaphandre v1.0 - Challenges and improvements to come on IT energy consumption evaluation
```
We get a YAML error otherwise

---
title: The road to Scaphandre v1.0 : Challenges and improvements to come on IT energy consumption evaluation
description: This blog post is about sharing what thoughts and insights we got during developing Scaphandre with our contributors, discussing with researchers and IT practitioners, regarding ways currently available to evaluate an IT server energy consumption and the limits of each. From this we try to picture some of the challenges ahead and leads we identify to work with them.
slug: 2023-the-road-to-scaphandre-v1-0-challenges-and-improvements-to-come-on-IT-energy-consumption-evaluation
weight: 1

> date:   2023-10-12 00:00:00 +0000

Reviewer: ```suggestion
date: 2023-10-13 00:00:00 +0000
```

author: Benoit Petit
---

> # The road to Scaphandre v1.0 : Challenges and improvements to come on IT energy consumption evaluation

Reviewer: ```suggestion
```
The title is set via the header

Those 2 ways share a common issue, we only get the global consumption of a machine. How to decide what actions to take to reduce the power used, beyond hardware changes, UEFI/BIOS configuration and firmware updates? How to get metrics from the infrastructures that would be useful to product and development teams and, moreover, everyone who is not working in the datacenter? Another question we might ask would be : how do we know the share of each component on the machine’s power? Here comes software-based power evaluation.

### Means and realms of evaluation

Let’s start with a visual overview of the different ways to evaluate a machine’s components energy consumption and the realm they cover.

> ![A map of ways to evaluate energy consumption of a server and its components](images/blogs/2023-10-the-road-to-scaphandre-v1.0-challenges-and-improvements-to-come-on-IT-energy-consumption-evaluation/map_it_energy_evaluation.webp "image_tooltip")

Reviewer: ```suggestion
<p class="mt-5 mb-5"><img src="images/blogs/2023-10-the-road-to-scaphandre-v1.0-challenges-and-improvements-to-come-on-IT-energy-consumption-evaluation/map_it_energy_evaluation.webp" alt="A map of ways to evaluate energy consumption of a server and its components"></p>
```

In the Windows world, each solution we have found that reads RAPL metrics (Intel Power Gadget and CPU-Z) have their own drivers. It’s needed to have a driver in kernel mode (which powercap and sysfs abstract us from, in the Linux world), as the __readmsr instruction is in kernel mode, so we wrote [one](https://github.com/hubblo-org/windows-rapl-driver/). For the record, MSR stands for Model Specific Register, those are the CPU registers storing RAPL metrics.

Once you get metrics from the driver, you also need to ensure you query those metrics for the right CPU-socket, cpu core-pinning is required if you want to address bi-socket or quad-socket machines.

> ![Comparison between GNU/Linux and Windows implementations of Scaphandre](images/blogs/2023-10-the-road-to-scaphandre-v1.0-challenges-and-improvements-to-come-on-IT-energy-consumption-evaluation/linux_vs_windows.webp)

Reviewer: ```suggestion
<p class="mt-5 mb-5"><img src="images/blogs/2023-10-the-road-to-scaphandre-v1.0-challenges-and-improvements-to-come-on-IT-energy-consumption-evaluation/linux_vs_windows.webp" alt="Comparison between GNU/Linux and Windows implementations of Scaphandre"></p>
```
